File: app/src/main/java/knf/kuma/tv/anime/FavPresenter.java
package knf.kuma.tv.anime;

import android.view.ViewGroup;

import androidx.leanback.widget.Presenter;
import knf.kuma.pojos.FavoriteObject;
import knf.kuma.tv.cards.FavCardView;

public class FavPresenter extends Presenter {

    public FavPresenter() {
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent) {
        return new ViewHolder(new FavCardView(parent.getContext()));
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, Object item) {
        ((FavCardView) viewHolder.view).bind((FavoriteObject) item);
    }

    @Override
    public void onUnbindViewHolder(ViewHolder viewHolder) {

    }
}

File: app/src/main/java/knf/kuma/tv/anime/FavPresenter.kt
package knf.kuma.tv.anime

import android.view.ViewGroup
import androidx.leanback.widget.Presenter
import knf.kuma.pojos.av1.FavoriteAV1
import knf.kuma.tv.cards.FavCardView

class FavPresenter : Presenter() {

    override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
        return ViewHolder(FavCardView(parent.context))
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, item: Any?) {
        if (item == null) return
        (viewHolder.view as FavCardView).bind(item as FavoriteAV1)
    }

    override fun onUnbindViewHolder(viewHolder: ViewHolder) {

    }
}
